apps/prairielearn/python/prairielearn.py

```diff
@@ -313,7 +313,7 @@ def to_json(v, *, df_encoding_version=1, np_encoding_version=1):
                 "_value": {
                     "index": list(v.index),
                     "columns": list(v.columns),
-                    "data": v.values.tolist(),
+                    "data": v.to_numpy().tolist(),
```

Comment:
I think this is covered by a unit test?

Reply:
What do you mean?

Reply:
Just want to double check this change is covered by a unit test, I think there is a test for this function.

Reply:
Yes:
```
@pytest.mark.parametrize(
    "df",
    [city_dataframe(), breast_cancer_dataframe(), r_types_dataframe()],
)
def test_encoding_legacy(df: pd.DataFrame) -> None:
    """Add compatibility test for legacy encoding"""
    reserialized_dataframe = cast(pd.DataFrame, pl.from_json(pl.to_json(df)))
    pd.testing.assert_frame_equal(df, reserialized_dataframe)
```
covers this branch.